pkg/tfgen/generate_nodejs.go

```diff
@@ -958,7 +958,14 @@ func tsPrimitive(vt schema.ValueType, elem interface{}, eleminfo *tfbridge.Schem
 		}
 		t = fmt.Sprintf("%s[]", elemType)
 	case schema.TypeMap:
-		t = fmt.Sprintf("{[key: string]: %v}", tsElemType(elem, eleminfo, out, wrapInput))
+		// If this map has a "resource" element type, just use the generated element type. This works around a bug in
+		// TF that effectively forces this behavior.
+		elemType := tsElemType(elem, eleminfo, out, wrapInput)
+		if _, hasResourceElem := elem.(*schema.Resource); hasResourceElem {
+			t, wrapInput = elemType, false
```

Comment:
Minor - for consistency with above, we could just `return elemType` here and drop the else (or change above to use a similar pattern as here).

Reply:
Ah, so we could. Done.